python/pydantic_core/_pydantic_core.pyi

```diff
@@ -355,6 +355,7 @@ def to_json(
     round_trip: bool = False,
     timedelta_mode: Literal['iso8601', 'float'] = 'iso8601',
     bytes_mode: Literal['utf8', 'base64'] = 'utf8',
+    inf_nan_mode: Literal['null', 'constants'] = 'null',
```

Comment:
To avoid a breaking change, I think the default needs to be:

```suggestion
    inf_nan_mode: Literal['null', 'constants'] = 'constants',
```

(Which is inconsistent with model config defaults, but I think in V3 we might change the model config default? We should check we have an issue to remember that.)